### PR TITLE
Secret GRPC Validation

### PR DESCRIPTION
**Pull request by tjholm**
Adapter layer validation for the secret service.

Will add additional tests for entire adapter layer once mocking infrastructure from secret plugin PRs is in.

### CODE REVIEW

File: pkg/plugins/secret/utils.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+	"github.com/nitric-dev/membrane/pkg/plugins"
+)
+
+const rx = "^\\w+(-\\w+)*$"
+
+// ValidateSecretName - Validates a secret name
+func ValidateSecretName(secName string) error {
+	if len(secName) == 0 {
+		return plugins.NewInvalidArgError("Secret name must not be blank")
+	}
```
Comment (medgar-nitric) [inline pkg/plugins/secret/utils.go:30]: Recommend error message:  `provide non-blank secret name`

File: pkg/plugins/secret/utils.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+	match, _ := regexp.MatchString(rx, secName)
+
+	if !match {
+		return plugins.NewInvalidArgError(fmt.Sprintf("Secret name must match pattern: %s", rx))
```
Comment (medgar-nitric) [inline pkg/plugins/secret/utils.go:35]: Recommend error message:  `provide secret name matching pattern: %s`

**Review by medgar-nitric — CHANGES_REQUESTED** [review]

File: pkg/plugins/secret/utils.go (hunk 1 above)
Comment (tjholm) [reply pkg/plugins/secret/utils.go:30]: done

File: pkg/plugins/secret/utils.go (hunk 2 above)
Comment (tjholm) [reply pkg/plugins/secret/utils.go]: done

**Review by jyecusch — APPROVED** [review]

**Review by medgar-nitric — APPROVED** [review]

**Pull request merged**